### PR TITLE
EPMDJ-2047. Fixed new build notification generation mechanism

### PR DESCRIPTION
**Pull request by s0larm00n**

### CODE REVIEW

File: plugin-api/drill-admin-part/src/jvmMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt [hunk 1]
```diff
@@ -15,7 +15,7 @@ abstract class AdminPluginPart<A>(
     abstract suspend fun processData(dm: DrillMessage): Any
     open suspend fun updateDataOnBuildConfigChange(buildVersion: String) = Unit
     open suspend fun applyPackagesChanges() = Unit
-    open suspend fun initialize() = Unit
+    open suspend fun initialize() = this
```
Comment (viktorbll) [inline plugin-api/drill-admin-part/src/jvmMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt]: This is totally unnecessary.

File: admin/core/src/main/kotlin/com/epam/drill/admindata/BuildManager.kt [hunk 2]
```diff
@@ -62,14 +55,21 @@ class AgentBuildManager(val agentId: String, val storeClient: StoreClient, lastB
...
             BcelClassParser(bytes, className).parseToJavaMethods()
         } ?: emptyMap()
         buildInfos[buildVersion] = buildInfos[buildVersion]?.copy(javaMethods = currentMethods)
             ?: BuildInfo(buildVersion = buildVersion)
         val buildInfo = buildInfos[buildVersion] ?: BuildInfo(buildVersion = buildVersion)
-        val prevMethods = buildInfos[buildInfo.prevBuild]?.javaMethods ?: mapOf()
+
+        val buildVersionIsNew = buildInfos[buildVersion] == null || buildInfos[buildVersion]!!.new
+        val prevBuild = if (buildVersionIsNew) lastBuild else buildInfo.prevBuild
+        val prevMethods = buildInfos[prevBuild]?.javaMethods ?: mapOf()
```
Comment (viktorbll) [inline admin/core/src/main/kotlin/com/epam/drill/admindata/BuildManager.kt]: emptyMap

File: plugin-api/drill-admin-part/src/jvmMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt (hunk 1 above)
Comment (s0larm00n) [reply plugin-api/drill-admin-part/src/jvmMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt]: agreed. originally done to beautify method instantiateAdminPluginPart(..) (AgentManager.kt, 292)
image

File: admin/core/src/main/kotlin/com/epam/drill/admindata/BuildManager.kt (hunk 2 above)
Comment (s0larm00n) [reply admin/core/src/main/kotlin/com/epam/drill/admindata/BuildManager.kt]: my bad, always the same mess

File: plugin-api/drill-admin-part/src/jvmMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt (hunk 1 above)
Comment (s0larm00n) [reply plugin-api/drill-admin-part/src/jvmMain/kotlin/com/epam/drill/plugin/api/end/AdminPluginPart.kt]: fixed in both requests

File: admin/core/src/main/kotlin/com/epam/drill/admindata/BuildManager.kt [hunk 3]
```diff
@@ -62,14 +55,21 @@ class AgentBuildManager(val agentId: String, val storeClient: StoreClient, lastB
         )
     }
 
-    suspend fun compareToPrev(buildVersion: String) {
+    suspend fun compareToPrev(
```
Comment (SunFellow) [inline admin/core/src/main/kotlin/com/epam/drill/admindata/BuildManager.kt]: Is `compareToPrev` a proper name here?
Usually `compare...` answers with some result of comparison. Here, it's apparently an operation that fills data structures basing on comparing builds and at the same time sends notifications. (Thus doing several things in one method). 
This refers to previous code as well, not to a change.

**Pull request closed without merging**